### PR TITLE
[new release] biocaml (0.11.2)

### PR DESCRIPTION
**Pull request by pveber**
The OCaml Bioinformatics Library

- Project page: <a href="https://github.com/biocaml/biocaml/">https://github.com/biocaml/biocaml/</a>

##### CHANGES:

* upgrade to core v0.15

### CODE REVIEW

File: packages/biocaml/biocaml.0.11.2/opam [hunk 1]
```diff
@@ -0,0 +1,56 @@
+opam-version: "2.0"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/biocaml/biocaml.0.11.2/opam:3]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/biocaml/biocaml.0.11.2/opam [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Biocaml aims to be a high-performance user-friendly library for
+Bioinformatics. See the [website](http://biocaml.org) for detailed API
+documentation. We welcome contributors and feedback from users. Please
+contact us with any comments and suggestions for features you would
+like added.
+"""
+depends: [
+  "base64"
+  "dune" {>= "1.10"}
+  "core" {>= "v0.15.0"}
```
Comment (mseri) [inline packages/biocaml/biocaml.0.11.2/opam:35]: Looks like moving to 0.15 breaks phylogenetics:
```
+ /home/opam/.opam/opam-init/hooks/sandbox.sh "build" "dune" "build" "-p" "phylogenetics" "-j" "31" "@install" (CWD=/home/opam/.opam/4.14/.opam-switch/build/phylogenetics.0.1.0)
- (cd _build/default/lib && /home/opam/.opam/4.14/bin/menhir newick_parser.mly --compile-errors newick_parser.messages) > _build/default/lib/newick_parser_errors.ml
- Read 3 sample input sentences and 2 error messages.
- File "lib/dune", line 6, characters 13-88:
- 6 |  (preprocess (pps ppx_deriving.show ppx_inline_test ppx_jane -allow-unannotated-ignores)))
-                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
- (cd _build/default && .ppx/225a9d522116ae1bff0c29e21b1aa610/ppx.exe -allow-unannotated-ignores --cookie 'inline_tests="disabled"' --cookie 'library-name="phylogenetics"' -o lib/newick_parser__mock.ml.pp.mock --impl lib/newick_parser__mock.ml.mock -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
- .ppx/225a9d522116ae1bff0c29e21b1aa610/ppx.exe: unknown option '-allow-unannotated-ignores'.
- ppx.exe [extra_args] [<files>]
-   -as-ppx                          Run as a -ppx rewriter (must be the first argument)
-   --as-ppx                         Same as -as-ppx
-   -as-pp                           Shorthand for: -dump-ast -embed-errors
-   --as-pp                          Same as -as-pp
-   -o <filename>                    Output file (use '-' for stdout)
```

You probably need to just add a conflict with ppx_jane > 0.15.0 there.

Comment (pveber) [reply packages/biocaml/biocaml.0.11.2/opam:35]: Thanks Marcello, this is fixed upstream, and I intended to submit a new release for `phylogenetics` as soon as this release of `biocaml` is published.

File: packages/biocaml/biocaml.0.11.2/opam [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  "camlzip" {>= "1.05"}
+  "xmlm"
+  "cfstream"
+  "ppx_compare"
+  "ppx_sexp_conv"
+  "re"
+  "rresult"
+  "uri"
+]
+depopts: ["async" "lwt"]
```
Comment (mseri) [inline packages/biocaml/biocaml.0.11.2/opam:47]:
```suggestion
depopts: ["async" "lwt"]
available: arch != "x86_32" & arch != "arm32"
```
due to
```
Use the camlp-streams library instead.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -short-paths -open Core -g -bin-annot -I lib/unix/.biocaml_unix.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/cfstream -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_expect/evaluator -I /home/opam/.opam/4.14/lib/ppx_expect/matcher -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/ppxlib/print_diff -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/re/perl -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/xmlm -I /home/opam/.opam/4.14/lib/zip -I lib/base/.biocaml_base.objs/byte -intf-suffix .ml -no-alias-deps -open Biocaml_unix -o lib/unix/.biocaml_unix.objs/byte/biocaml_unix__Bam.cmo -c -impl lib/unix/bam.pp.ml)
# File "lib/unix/bam.ml", line 627, characters 13-23:
# 627 |       | Some 2147483647 (* POS in BAM is 0-based *)
#                    ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -short-paths -open Core -g -bin-annot -I lib/unix/.biocaml_unix.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/cfstream -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_expect/evaluator -I /home/opam/.opam/4.14/lib/ppx_expect/matcher -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/ppxlib/print_diff -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/re/perl -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/xmlm -I /home/opam/.opam/4.14/lib/zip -I lib/base/.biocaml_base.objs/byte -intf-suffix .ml -no-alias-deps -open Biocaml_unix -o lib/unix/.biocaml_unix.objs/byte/biocaml_unix__Sam.cmo -c -impl lib/unix/sam.pp.ml)
# File "lib/unix/sam.ml", line 216, characters 34-44:
# 216 |   (if (1 <= length) && (length <= 2147483647) then
#                                         ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -short-paths -open Core -g -I lib/unix/.biocaml_unix.objs/byte -I lib/unix/.biocaml_unix.objs/native -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/cfstream -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_expect/evaluator -I /home/opam/.opam/4.14/lib/ppx_expect/matcher -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/ppxlib/print_diff -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/re/perl -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/xmlm -I /home/opam/.opam/4.14/lib/zip -I lib/base/.biocaml_base.objs/byte -I lib/base/.biocaml_base.objs/native -intf-suffix .ml -no-alias-deps -open Biocaml_unix -o lib/unix/.biocaml_unix.objs/native/biocaml_unix__Sam.cmx -c -impl lib/unix/sam.pp.ml)
# File "lib/unix/sam.ml", line 216, characters 34-44:
# 216 |   (if (1 <= length) && (length <= 2147483647) then
#                                         ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
```

File: packages/biocaml/biocaml.0.11.2/opam (hunk 2 above)
Comment (mseri) [reply packages/biocaml/biocaml.0.11.2/opam:35]: This is fixed in a separate PR

File: packages/biocaml/biocaml.0.11.2/opam (hunk 3 above)
Comment (pveber) [reply packages/biocaml/biocaml.0.11.2/opam:47]: Thanks, I've propagated the constraint upstream!

**Pull request merged**